res/controllers/DJControl-Compact-scripts.js

```diff
@@ -0,0 +1,113 @@
+// Mode button does give out midi, but the selection itself is done inside the
+// controller -- no control over which mode is selected for instance.
+
+HercDJCompact = new function() {
+   this.group = "[Master]";
+}
+
+HercDJCompact.init = function(id) {
+    scratch = false;
+    scratch_timer = [];
+    scratch_timer_on = [];
+
+    engine.connectControl("[Recording]", "status",
+            "HercDJCompact.OnRecordingStatusChange");
+
+    // Tell controller to send midi to update knob and slider positions.
+    midi.sendShortMsg(0xB0, 0x7F, 0x7F);
```

Comment:
my vestax VCI400 just does it on connect, for instance, no midi trigger required.  It's certainly not any sort of standard, so I'd guess that someone at Hercules thought it up and they reuse firmwares on multiple devices.